src/identity/Identity.ts

```diff
@@ -131,28 +148,43 @@ export default class Identity extends PublicIdentity {
    * Identity.buildFromURI('//Bob');
    * ```
    */
-  public static buildFromURI(uri: string): Identity {
+  public static async buildFromURI(uri: string): Promise<Identity> {
     const keyring = new Keyring({ type: 'ed25519' })
     const derived = keyring.createFromUri(uri)
-    // TODO: heck to create identity from //Alice
-    return new Identity(u8aUtil.u8aToU8a(uri), derived)
+    const seed = u8aUtil.u8aToU8a(uri)
+
+    // pad seed if needed for claimer
+    let paddedSeed = u8aUtil.u8aToU8a([...seed])
```

Comment:
DRY

Reply:
fixed in a4a4b571a3299dce5fcbf18b3855e7313f306862